lib/rubocop/cop/style/collection_compact.rb

```diff
@@ -8,7 +8,8 @@ module Style
       #
       # @safety
       #   It is unsafe by default because false positives may occur in the
-      #   `nil` check of block arguments to the receiver object.
+      #   `nil` check of block arguments to the receiver object. Additionally,
+      #    .compact may not exist for the highlighted object.
```

Comment:
I'd say something like "we can't know for certain the type of the receiver object, which may result in false positives as well".

Reply:
Will do! Thanks for the quick response @bbatsov, I appreciate it :)